### PR TITLE
Introduce and favor an exactperk: search

### LINKED ISSUE #3486: Search by specific perk
I'd like to make it so we autocomplete known perks, and when you select from the autocomplete it just searches for the exact hash you're looking for.

### PR DESCRIPTION
**Pull request by bhollis**
Fixes #3486, and probably gets rid of a few annoying edge cases in Compare and Organizer. It's a bit awkwardly named but probably better not to change the meaning of an existing search?

### CODE REVIEW

File: src/app/search/search-filters/freeform.ts [hunk 1]
```diff
@@ -150,17 +150,20 @@ const freeformFilters: FilterDefinition[] = [
...
     },
-    filter: ({ filterValue, language, d2Definitions }) => {
-      const startWord = startWordRegexp(plainString(filterValue, language), language);
-      const test = (s: string) => startWord.test(plainString(s, language));
+    filter: ({ lhs, filterValue, language, d2Definitions }) => {
+      const normalized = plainString(filterValue, language);
+      let test = (s: string) => normalized === plainString(s, language);
+      if (lhs === 'perkname') {
+        test = (s: string) => normalized === plainString(s, language);
+      }
```
Comment (robojumper) [inline src/app/search/search-filters/freeform.ts:162]: These two predicates are identical, so `perkname` also gets the new behavior.

Comment (bhollis) [reply src/app/search/search-filters/freeform.ts]: Whoops, not sure how that happened

File: config/i18n.json [hunk 2]
```diff
@@ -279,7 +279,7 @@
     "PartialMatch": "Shows items where their name, description, any perk, or any mod has a partial match to the filter text. Search for entire phrases using quotes. the 'perk:' filter will only search perks.",
     "PatternUnlocked": "Shows items that have a crafting pattern unlocked, even if the item itself isn't crafted.",
     "Perk": "Shows items where one of their perks or mods has a partial match to the filter text in their name or description. Search for entire phrases using quotes.",
-    "PerkName": "Shows items where one of their perks or mods has a partial match to the filter text in their name only. Search for entire phrases using quotes.",
+    "PerkName": "Shows items where one of their perks' or mods' names matches (exactperk:) or partially matches (perkname:) the filter text. Search for entire phrases using quotes.",
```
Comment (nev-r) [inline config/i18n.json:282]:
```suggestion
    "PerkName": "Shows items with a perk or mod whose name matches (exactperk:) or partially matches (perkname:) the filter text. Search for entire phrases using quotes.",
```

File: src/locale/en.json [hunk 3]
```diff
@@ -279,7 +279,7 @@
     "PartialMatch": "Shows items where their name, description, any perk, or any mod has a partial match to the filter text. Search for entire phrases using quotes. the 'perk:' filter will only search perks.",
     "PatternUnlocked": "Shows items that have a crafting pattern unlocked, even if the item itself isn't crafted.",
     "Perk": "Shows items where one of their perks or mods has a partial match to the filter text in their name or description. Search for entire phrases using quotes.",
-    "PerkName": "Shows items where one of their perks or mods has a partial match to the filter text in their name only. Search for entire phrases using quotes.",
+    "PerkName": "Shows items where one of their perks' or mods' names matches (exactperk:) or partially matches (perkname:) the filter text. Search for entire phrases using quotes.",
```
Comment (nev-r) [inline src/locale/en.json:282]:
```suggestion
    "PerkName": "Shows items with a perk or mod whose name matches (exactperk:) or partially matches (perkname:) the filter text. Search for entire phrases using quotes.",
```

**Pull request merged**